src/language-js/print/flow.js

```diff
@@ -84,6 +84,10 @@ function printFlow(path, options, print) {
       ];
     case "IndexedAccessType":
       return printIndexedAccessType(path, options, print);
+    case "OptionalIndexedAccessType":
+      return node.optional
+        ? [print("objectType"), "?.", "[", print("indexType"), "]"]
+        : printIndexedAccessType(path, options, print);
```

Comment:
Let's move the `optional` check to function, so we can keep this file small.

Reply:
Only thing is that `printIndexedAccessType` is shared between Flow and TypeScript, and while both of them have indexed access types, but only Flow supports optional indexed access types. Is that OK?

Reply:
Yes, we can use `node.type === 'OptionalIndexedAccessType' && node.optional`, make it strict.